server/src/main/java/org/opensearch/index/search/QueryStringQueryParser.java

```diff
@@ -642,20 +670,24 @@ private Query getPossiblyAnalyzedPrefixQuery(String field, String termStr, Mappe
             Query posQuery;
             if (plist.size() == 1) {
                 if (isLastPos) {
+                    onDiscoveredField(currentFieldType.name());
                     posQuery = currentFieldType.prefixQuery(plist.get(0), getMultiTermRewriteMethod(), context);
                 } else {
+                    onDiscoveredField(field);
                     posQuery = newTermQuery(new Term(field, plist.get(0)), BoostAttribute.DEFAULT_BOOST);
                 }
             } else if (isLastPos == false) {
                 // build a synonym query for terms in the same position.
                 SynonymQuery.Builder sb = new SynonymQuery.Builder(field);
                 for (String synonym : plist) {
+                    onDiscoveredField(field);
                     sb.addTerm(new Term(field, synonym));
                 }
                 posQuery = sb.build();
             } else {
                 List<BooleanClause> innerClauses = new ArrayList<>();
                 for (String token : plist) {
+                    onDiscoveredField(field);
```

Comment:
Same, the field does not change for a token, it should be done only once before

Reply:
done